<a href="https://colab.research.google.com/github/cleogrs/analytics/blob/main/C%C3%B3pia_de_CleoGomesdaRochaSantos_desafiopythonparadados.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Cléo Gomes da Rocha Santos - Turma 6 do Bootcamp Data Analytics Womakerscode

In [ ]:
import pandas as pd
import numpy as np

#importando csv
df = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/saude_do_sono_estilo_vida.csv')

#cabeçalho
df.head()


ID Gênero  Idade                 Ocupação  Duração do sono  \
0   1  Homem     27         Eng. de Software              6.1   
1   2  Homem     28                Médico(a)              6.2   
2   3  Homem     28                Médico(a)              6.2   
3   4  Homem     28  Representante de Vendas              5.9   
4   5  Homem     28  Representante de Vendas              5.9   

   Qualidade do sono  Nível de atividade física  Nível de estresse  \
0                  6                         42                  6   
1                  6                         60                  8   
2                  6                         60                  8   
3                  4                         30                  8   
4                  4                         30                  8   

  Categoria BMI Pressão sanguíneaaaa  Frequência cardíaca  Passos diários  \
0     Sobrepeso               126/83                   77            4200   
1        Normal               125/80                   75           10000   
2        Normal               125/80                   75           10000   
3     Obesidade               140/90                   85            3000   
4     Obesidade               140/90                   85            3000   

  Distúrbio do sono  
0           Nenhuma  
1           Nenhuma  
2           Nenhuma  
3    Apneia do sono  
4    Apneia do sono

1. Ao visualizar a base você percebeu que seria melhor alterar o nome de algumas colunas. Mude o ‘ID’ para ‘Identificador’, corrija o nome da coluna que indica a pressão sanguínea, mude a coluna ‘Ocupação’ para ‘Profissão’, a coluna ‘Categoria BMI’ está em parte em inglês, substitua para ‘Categoria IMC’.


In [ ]:
# Renomeando a coluna ID
df.rename(columns={'ID': 'Identificador', 'Pressão sanguíneaaaa': 'Pressão sanguínea', 'Ocupação': 'Profissão', 'Categoria BMI': 'Categoria IMC'}, inplace=True)
df.head()

Identificador Gênero  Idade                Profissão  Duração do sono  \
0              1  Homem     27         Eng. de Software              6.1   
1              2  Homem     28                Médico(a)              6.2   
2              3  Homem     28                Médico(a)              6.2   
3              4  Homem     28  Representante de Vendas              5.9   
4              5  Homem     28  Representante de Vendas              5.9   

   Qualidade do sono  Nível de atividade física  Nível de estresse  \
0                  6                         42                  6   
1                  6                         60                  8   
2                  6                         60                  8   
3                  4                         30                  8   
4                  4                         30                  8   

  Categoria IMC Pressão sanguínea  Frequência cardíaca  Passos diários  \
0     Sobrepeso            126/83                   77            4200   
1        Normal            125/80                   75           10000   
2        Normal            125/80                   75           10000   
3     Obesidade            140/90                   85            3000   
4     Obesidade            140/90                   85            3000   

  Distúrbio do sono  
0           Nenhuma  
1           Nenhuma  
2           Nenhuma  
3    Apneia do sono  
4    Apneia do sono

2. Qual é a média, a moda e a mediana de horas de sono para cada uma
das profissões? [‘mean’, np.median, pd.Series.mod]

In [ ]:
#soma das horas de sono por profissões
aggprofs = df.groupby('Profissão')['Duração do sono'].agg(['mean', np.median, pd.Series.mode])
aggprofs

mean  median        mode
Profissão                                            
Advogado(a)              7.410638     7.3         7.2
Cientista                6.000000     6.0  [5.8, 6.2]
Contador(a)              7.113514     7.2         7.2
Enfermeiro(a)            7.048611     6.5         6.1
Eng. de Software         6.750000     6.8         7.5
Engenheiro(a)            7.987302     8.3         8.4
Gerente                  6.900000     6.9         6.9
Médico(a)                6.970423     7.6         6.0
Pessoa Vendendora        6.403125     6.4         6.5
Professor(a)             6.690000     6.6         6.6
Representante de Vendas  5.900000     5.9         5.9

3. Das pessoas que atuam com engenharia de software qual a
porcentagem de obesos?

In [ ]:
#filtrando o dataframe para engenheiros de software
engsoft = df[df['Profissão'] == 'Eng. de Software']

#frequência absoluta IMC
fr_absoluta_engsoftob = pd.crosstab(engsoft['Profissão'], engsoft['Categoria IMC'])

#aplicação da fórmula
pc_engsoftob = fr_absoluta_engsoftob.div(fr_absoluta_engsoftob.sum(1), axis=0) * 100
pc_engsoftob

Categoria IMC     Obesidade  Peso normal  Sobrepeso
Profissão                                          
Eng. de Software       25.0         50.0       25.0

**R:A porcentagem de Engenheiros de Software com Obesidade é de 25%**

4. De acordo com os dados, advogar ou ser representante de vendas faz
você dormir menos? (Use o método ‘isin’, considere a média)

In [ ]:
#definindo a média dos valores de duração de sono de todas as profissões
df['Duração do sono'].mean()

7.129490616621984

In [ ]:
#filtrando o dataframe para advogados
df_advogados = df[df['Profissão'].isin(['Advogado(a)'])]
#cálculo da média da duração do sono para advogados
media_advogados = df_advogados['Duração do sono'].mean()
media_advogados

7.4106382978723415

In [ ]:
#filtrando o dataframe para representantes de vendas
df_representantes = df[df['Profissão'].isin(['Representante de Vendas'])]
df_representantes
#cálculo da média da duração do sono para representante de vendas
media_representantes = df_representantes['Duração do sono'].mean()
media_representantes

5.9

In [ ]:
#filtrando o dataframe para advogados e representantes de vendas
df_advogados_representantes = df[df['Profissão'].isin(['Advogado(a)', 'Representante de Vendas'])]
#cálculo da média da duração do sono para advogados e representante de vendas
media_advogados_representantes = df_advogados_representantes['Duração do sono'].mean()
media_advogados_representantes

7.348979591836734

**R: Considerando a média das horas de sono:**
1. **Somente Advogados dormem mais que a média das outras profissões**
2. **Somente Representantes de vendas dormem menos do que a média das outras profissões**
3. **Sendo advogado ou representante de venda faz você dormir mais que a média das profissões**

5. Entre quem fez enfermagem e quem fez medicina, quem tem menos horas de sono? (Use o método ‘isin’, considere a média)

In [ ]:
#filtrando o dataframe para enfermeiros
df_enfs = df[df['Profissão'].isin(['Enfermeiro(a)'])]
#cálculo da média da duração do sono para enfermeiros
media_enfs = df_enfs['Duração do sono'].mean()
media_enfs

7.048611111111111

In [ ]:
#filtrando o dataframe para médicos
df_med = df[df['Profissão'].isin(['Médico(a)'])]
#cálculo da média da duração do sono para médicos
media_med = df_med['Duração do sono'].mean()
media_med

6.970422535211269

**R:Considerando a média das profissões, Médicos(as) dormem menos do que Enfermeiros(as).**

6. Faça um subconjunto com as colunas Identificador, Gênero, Idade,
Pressão sanguínea e Frequência cardíaca.

In [ ]:
dfsub = df[['Identificador', 'Gênero', 'Idade', 'Pressão sanguínea', 'Frequência cardíaca']]
dfsub

Identificador  Gênero  Idade Pressão sanguínea  Frequência cardíaca
0                1   Homem     27            126/83                   77
1                2   Homem     28            125/80                   75
2                3   Homem     28            125/80                   75
3                4   Homem     28            140/90                   85
4                5   Homem     28            140/90                   85
..             ...     ...    ...               ...                  ...
368            369  Mulher     59            140/95                   68
369            370  Mulher     59            140/95                   68
370            371  Mulher     59            140/95                   68
371            372  Mulher     59            140/95                   68
372            373  Mulher     59            140/95                   68

[373 rows x 5 columns]

7. Descubra qual a profissão menos frequente no conjunto. (Use
value_counts)

In [ ]:
sumprofs = df['Profissão'].agg(['value_counts'])
sumprofs

value_counts
Enfermeiro(a)                      72
Médico(a)                          71
Engenheiro(a)                      63
Advogado(a)                        47
Professor(a)                       40
Contador(a)                        37
Pessoa Vendendora                  32
Eng. de Software                    4
Cientista                           4
Representante de Vendas             2
Gerente                             1

**R: A profissão menos frequente é Gerente.**

8. Quem tem maior pressão sanguínea média, homens ou mulheres?
(Considere a média)

In [ ]:
#separando valores de pressão sistólica e diastólica pelo delimitador
pa = df['Pressão sanguínea'].str.split("/")
pas = pa.str.get(0)
pad = pa.str.get(1)

#novas colunas
df['Pressão sistólica'] = pas
df['Pressão diastólica'] = pad

#convertendo para float
df['Pressão sistólica'] = df['Pressão sistólica'].astype(float)
df['Pressão diastólica'] = df['Pressão diastólica'].astype(float)

#filtrando dataframe para mulheres
media_pa = df.groupby('Gênero')[['Pressão sistólica','Pressão diastólica']].mean().round()
media_pa

Pressão sistólica  Pressão diastólica
Gênero                                       
Homem               127.0                83.0
Mulher              130.0                86.0

**R: As mulheres possuem uma pressão sanguínea média maior (aproximadamente 130/86) do que os homens (aproximadamente 127/83).**

9. É predominante entre os participantes dormir 8 horas por dia
(considere usar Moda como medida)?

In [ ]:
#moda não arredondada
sonomoda1 = df['Duração do sono'].mode()
sonomoda1

0    7.2
Name: Duração do sono, dtype: float64

In [ ]:
#convertendo para float
df['Duração do sono'] = df['Duração do sono'].astype(float)
#arredondando valores
df['Duração do sono'] = df['Duração do sono'].round()
#moda arredondada das horas de sono
sonomoda2 = df['Duração do sono'].agg([pd.Series.mode])
sonomoda

mode
0   6.0

**R: Não é predominante entre os participantes dormir 8 horas por dia. Foram avaliadas as modas com números decimais e arredondados, e em nenhum dos resultados demonstrou que 8 horas é predominante.**

10. Pessoas com frequências cardíacas acima de 70 dão mais passos que
pessoas com frequência cardíaca menor ou igual a 70? (Use a média)

In [ ]:
#filtrando o dataframe para pessoas com frequencia maior que 70
freq_maior70 = df[df['Frequência cardíaca']>70]

#realizando média
cross_freq_passos_maior70 = freq_maior70.groupby('Frequência cardíaca')['Passos diários'].sum().mean().round()
cross_freq_passos_maior70

61493.0

In [ ]:
#filtrando o dataframe para pessoas com frequencia menor ou igual a 70
freq_igualmenor70 = df[df['Frequência cardíaca']<=70]

#realizando média
cross_freq_passos_menor = freq_igualmenor70.groupby('Frequência cardíaca')['Passos diários'].sum().mean().round()
cross_freq_passos_menor

336320.0

**R: Pessoas com frequência cardíaca menor ou igual a 70 dão, em média, mais passos (336320) do que pessoas com frequência cardíaca acima de 70 (61493 passos).**